### ISSUE TITLE
How can i get character scores(or confidences) in Trocr?

### ISSUE THREAD

**Issue by dcdethan** [NielsRogge/Transformers-Tutorials#192]
Hi,thank you for your great job ,now i have a question, How can i get character scores(or confidences) in Trocr? i according this file--"root/src/transformers/models/trocr/convert_trocr_unilm_to_pytorch.py" to eval.

**Comment by dcdethan** [comment]
Now,i only get first character scores by code:
'''
pixel_values = self.processor([img], return_tensors="pt").pixel_values
decoder_input_ids = torch.tensor([[model.config.decoder.decoder_start_token_id]])
outputs = model(pixel_values=pixel_values, decoder_input_ids=decoder_input_ids)
logits = outputs.logits
first_index = logits.argmax(-1).item()
'''
how i get all character scores?

**Comment by NielsRogge** [comment]
You can call the [generate](https://huggingface.co/docs/transformers/v4.22.2/en/main_classes/text_generation#transformers.generation_utils.GenerationMixin.generate) method (as shown in the notebooks) and specify `return_scores=True`.

**Comment by dcdethan** [comment]
> You can call the [generate](https://huggingface.co/docs/transformers/v4.22.2/en/main_classes/text_generation#transformers.generation_utils.GenerationMixin.generate) method (as shown in the notebooks) and specify `return_scores=True`.

Thank you for your reply. I specify output_scores=True and return_dict_in_generate=True,it works.

**Comment by tairen99** [comment]
Hi @dcdethan and @NielsRogge ,

Thank you for your comments on the word level of confidence.

I also get confidence using the same options, but the maximum confidence is about 30. 

Are you getting the same level of confidence? if not, how do we get the right confidence?

Thank you in advance!
